provider/kubernetes_test.go

```diff
@@ -1524,11 +1524,80 @@ func TestServiceAnnotations(t *testing.T) {
 	}
 }
 
+func TestGetServiceErrors(t *testing.T) {
+	ingresses := []*v1beta1.Ingress{
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Namespace: "testing",
+				Annotations: map[string]string{
+					"traefik.frontend.passHostHeader": "true",
+				},
+			},
+			Spec: v1beta1.IngressSpec{
+				Rules: []v1beta1.IngressRule{
+					{
+						Host: "foo",
+						IngressRuleValue: v1beta1.IngressRuleValue{
+							HTTP: &v1beta1.HTTPIngressRuleValue{
+								Paths: []v1beta1.HTTPIngressPath{
+									{
+										Path: "/bar",
+										Backend: v1beta1.IngressBackend{
+											ServiceName: "notExists",
+											ServicePort: intstr.FromInt(80),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	services := []*v1.Service{}
+	endpoints := []*v1.Endpoints{}
+	watchChan := make(chan interface{})
+
+	client := clientMock{
+		ingresses:       ingresses,
+		services:        services,
+		endpoints:       endpoints,
+		watchChan:       watchChan,
+		apiServiceError: true,
+	}
+	provider := Kubernetes{}
+	_, err := provider.loadIngresses(client)
+
+	if err == nil {
+		t.Fatal("Should have raised error when loading services but did not.")
+	}
+
+	client = clientMock{
+		ingresses:         ingresses,
+		services:          services,
+		endpoints:         endpoints,
+		watchChan:         watchChan,
+		apiEndpointsError: true,
+	}
+	provider = Kubernetes{}
+	_, err = provider.loadIngresses(client)
```

Comment:
I suggest we convert the test into a [table-driven one](https://github.com/golang/go/wiki/TableDrivenTests) using [sub-tests](https://blog.golang.org/subtests). Among several benefits, this should improve readability and reduce the amount of test code needed.